### PR TITLE
Allows setting fixed special scoped parameters for start locations

### LINKED ISSUE #72479: Start location parameters
### Is your feature request related to a problem? Please describe.

Parameters allow heavily simplifying variations on mapgen by reusing maps with different palettes without duplicating them, allowing the selection of multiple nests over a special to allow for pseudo z-level and pseudo nests greater than 24x24 and more.

Currently you can't specify a start location to pick or only use OMTs with a chosen parameter.

Specifically I'm simplifying the evac centre maps down to a few OMTs but there are various start locations expecting specific variants (safe/infested/vandalised) that are blocking it.

This would also allow for having 0 weight parameters to be selected for start locations only which would be nice.

### Solution you would like.

I think the start location overriding the random distribution and setting a parameter itself makes most sense.

Example syntax:
```
  {
    "type": "start_location",
    "id": "sloc_shelter_infested",
    "name": "Infested Evac Center",
    "terrain": [
      { "om_terrain": "shelter", "parameters": { "palette": "shelter_infested" } }
    ],
  },
```
where palette is a parameter id specified in the shelter palette:
```
    "parameters": {
      "palette": {
        "type": "palette_id",
        "default": {
          "distribution": [
            "shelter_basic",
            "shelter_survivor",
            "shelter_infested",
            "shelter_vandal",
            "shelter_used"
          ]
        }
      }
    },
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Procyonae**
#### Summary
None

#### Purpose of change

Closes #72479
Needed for #72561 and #72809

#### Describe the solution

You can set singular or multiple parameters to singular values which then override those calculated normally when the mapdata is initialised, so you don't have to set every parameter. 
You can also use this to set a parameter to a 0 weight value for it to be start location only. 

set_parameters somewhat cargo cults overmap_ui's set_special_args used for debug setting parameters

#### Describe alternatives you've considered

The bool being stored in overmap_buffer is pretty gross but I couldn't see how to easily scope it otherwise

It still calcs the overridden parameters which is suboptimal but I didn't fancy attacking the function that spits them all out together

If I or someone else gets around to doing #72529 the functions will probably want moving to overmap_special but I didn't fancy trying to scope that for this

#### Testing

Tested with #72561

### CODE REVIEW

File: src/start_location.h [hunk 1]
```diff
@@ -27,6 +27,12 @@ struct start_location_placement_constraints {
     numeric_interval<int> allowed_z_levels{ -OVERMAP_DEPTH, OVERMAP_HEIGHT };
 };
 
+struct omt_types_parameters {
+    std::string omt;
+    ot_match_type omt_type;
+    std::vector<std::pair<std::string, std::string>> parameters;
```
Comment (mqrause) [inline src/start_location.h:33]: Is there a reason to use a vector of pairs instead of a map here?

Comment (Procyonae) [reply src/start_location.h]: To begin with I were relying on iterators to match positions between two vectors for some reason but ye I don't think there's any need

**Comment by Maleclypse** [comment]
As an in repo use case of this `You can also use this to set a parameter to a 0 weight value for it to be start location only.` can you have the Ready or Not Mansion start with special ferals only exist in that start?   

Sorry to request some extra work from you but I think that will both demonstrate a usecase and close a stable blocker.

**Comment by Procyonae** [comment]
Opened the mansion feral PR #72695

**Comment by Procyonae** [comment]
> As an in repo use case of this `You can also use this to set a parameter to a 0 weight value for it to be start location only.` can you have the Ready or Not Mansion start with special ferals only exist in that start?
> 
> Sorry to request some extra work from you but I think that will both demonstrate a usecase and close a stable blocker.

Is there any issue with using a draft PR as a use case? I'd of put this in with the evac centre PR but I thought it made more sense to separate them

**Pull request merged**